### ISSUE TITLE
Consider switching to the out-of-tree AWS cloud-controller-manager

### ISSUE THREAD

**Issue by ialidzhikov** [gardener/gardener-extension-provider-aws#577]
**How to categorize this issue?**

/area control-plane
/kind enhancement
/platform aws

**What would you like to be added**:
Since some time https://github.com/kubernetes/cloud-provider-aws has releases (that are not alpha or beta) and pushes container images.

We can check the compatibility with the in-tree CCM and, if possible, switch to the out-of-tree AWS CCM. We could do this by deploying the OOT CCM staring given K8s version (for example K8s 1.25).

**Why is this needed**:
To get rid of our in-tree CCM fork (https://github.com/gardener/cloud-provider-aws) and all the maintenance effort related to it.

**Comment by dkistner** [comment]
Yes I fully agree. Finally it seems that the is also a published AWS ccm that we can consume. Adopting it with `v1.25` sounds reasonable for me (after successful validation of course).
